setup_monitoring.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+######################################################################################
+## Install Prometheus operator
+######################################################################################
+function install_prometheus_operator() {
+  echo "📦 installing the Prometheus operator..."
+  export KUBECONFIG=$KUBECONFIG_HOST
+
+  oc process -f config/monitoring/prometheus.tmpl.yaml \
+    -p NAMESPACE="$TOOLCHAIN_OPERATOR_NS_HOST" \
+    -p SA_NAME=prometheus-k8s \
+    | oc apply -f -
+  
+  # needed by the `kube-rbac-proxy` sidecar to create TokenReviews and SubjectAccessReviews
+  oc adm policy add-cluster-role-to-user kube-rbac-proxy -z prometheus-k8s -n "$TOOLCHAIN_OPERATOR_NS_HOST"
+  
+  echo "✅ done installing Prometheus operator"
+  echo ""
+}
+
+######################################################################################
+## Deploying Grafana
+######################################################################################
+function deploy_grafana() {
+  echo "📦 deploying Grafana..."
+  # fetch the route to Thanos Querier on member cluster and 
+  export KUBECONFIG=$KUBECONFIG_MEMBER
+  THANOS_QUERIER_URL_MEMBER="https://$(oc get route/thanos-querier -n openshift-monitoring -o json | jq -r '.status.ingress[0].host')"
+  export THANOS_QUERIER_URL_MEMBER
+  echo "route to Thanos Querier on member cluster: $THANOS_QUERIER_URL_MEMBER"
+  
+  # create a `grafana` serviceaccount which is allowed to access Prometheus (via kube-rbac-proxy)
+  oc process -f config/monitoring/grafana_serviceaccount.tmpl.yaml \
+    -p NAMESPACE="$TOOLCHAIN_OPERATOR_NS_MEMBER" \
+    -p NAME=grafana \
```

Comment:
hmm... I probably got lost - you used `prometheus-k8s` SA when you installed the prometheus operator :thinking: 
https://github.com/codeready-toolchain/toolchain-infra/pull/43/files#diff-d58fd87c8b007fafd44f1fc3cc0ed5b991835609b837d86c85c8a1b6cd3d5a6fR16
```
 oc adm policy add-cluster-role-to-user kube-rbac-proxy -z prometheus-k8s -n "$TOOLCHAIN_OPERATOR_NS_HOST"
```

Reply:
ok, so we have 2 service accounts:
- `prometheus-k8s` is used to run Prometheus and it needs some permissions to create `subjectaccessreviews` and `tokenreviews` (the `kube-rbac-proxy` sidecar checks the token in the incoming requests)
- `grafana` is a token used by Grafana to request data on Prometheus. This SA needs some RBAC permissions, which will be checked by the `kube-rbac-proxy` sidecar in the Prometheus pod. This SA is not used to run any application, only to authn/authz on Prometheus via HTTP requests.

I hope this clarifies!

Reply:
cool, thanks a lot for clarification :+1: